pkg/server/endpoints/authorized_entryfetcher.go

```diff
@@ -115,28 +122,62 @@ func (a *AuthorizedEntryFetcherWithEventsBasedCache) updateRegistrationEntriesCa
 
 	seenMap := map[string]struct{}{}
 	for _, event := range resp.Events {
+		// If there is a gap in the event log the missed events for later processing
+		if event.EventID != a.lastRegistrationEntryEventID+1 {
+			for i := a.lastRegistrationEntryEventID + 1; i < event.EventID; i++ {
+				a.missedRegistrationEntryEvents[i] = struct{}{}
+			}
+		}
+
 		// Skip fetching entries we've already fetched this call
 		if _, seen := seenMap[event.EntryID]; seen {
 			a.lastRegistrationEntryEventID = event.EventID
```

Comment:
I believe that a small bug might exist here too, as it is not clear to me that the EventID is always going to be greater than a.lastRegistreationEntryEventID.  Perhaps make the assignment only after comparing if EventID is greater?

Reply:
We get the list of events in ascending order

Reply:
That's good enough for me.